annotations/views.py

```diff
@@ -685,14 +685,46 @@ def add_text_to_collection(request, *args, **kwargs):
     return JsonResponse({})
 
 
-@login_required
 def user_details(request, userid, *args, **kwargs):
+    """
+    Provides users with their own profile view and public profile view of other users in case they are loggedIn.
+    Provides users with public profile page in case they are not loggedIn
+    ----------
+    request : HTTPRequest
+        The request for fetching user details
+    userid : int
+        The userid of user who's data  needs to be fetched
+    args : list
+        List of arguments to view
+    kwargs : dict
+        dict of arugments to view
+    Returns
+    ----------
+    :HTTPResponse:
+        Renders an user details view based on user's authentication status.
+    """
+
     user = get_object_or_404(VogonUser, pk=userid)
-    template = loader.get_template('annotations/user_details.html')
-    context = RequestContext(request, {
-        'user': request.user,
-        'detail_user': user,
-    })
+    if request.user.is_authenticated() and request.user.id == userid:
+        template = loader.get_template('annotations/user_details.html')
+        context = RequestContext(request, {
+            'user': request.user,
+            'detail_user': user,
+        })
+    else:
+        textCount = Text.objects.filter(addedBy=user).count()
+        annotationsSet = Text.objects.filter(annotators=user).distinct()
+        annotationCount = 0
+        for text in annotationsSet:
+            annotationCount += text.annotationCount
```

Comment:
We can simply this quite a bit by using the ``RelatedManager`` for ``Annotations`` on the ``VogonUser`` class. Take a look at the ``Annotation`` model: since there is a ``ForeignKey`` to ``VogonUser`` with no explicit ``related_name``, there should automatically be a ``RelatedManager`` on ``VogonUser`` called ``annotation_set``. So instead of iterating over all of the texts that the user has annotated, you can just do:

```python
annotationCount = user.annotation_set.count()
```

That **should** work, anyway. Give it a try!